.github/workflows/ci.yml

```diff
@@ -13,7 +13,7 @@ env:
   CARGO_PROFILE_TEST_DEBUG: 0
   CARGO_PROFILE_DEV_DEBUG: 0
   # If nightly is breaking CI, modify this variable to target a specific nightly version.
-  NIGHTLY_TOOLCHAIN: nightly-2025-05-16 # pinned until a fix for https://github.com/rust-lang/miri/issues/4323 is released
+  NIGHTLY_TOOLCHAIN: nightly-2025-05-14 # pinned until a fix for https://github.com/rust-lang/miri/issues/4323 is released
```

Comment:
Do not merge like this, I tested the Miri issues on this branch. I will remove these changes.